__tests__/waveHandPicker.test.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { WaveHandPicker } from "@/components/WaveHandPicker";
+import "@testing-library/jest-dom";
+
+describe("WaveHandPicker", () => {
+  const mockUpdateHandWaveBadge = jest.fn();
+  const mockOnRetry = jest.fn();
+  const mockOnAdd = jest.fn();
+  const mockOnDelete = jest.fn();
+
+  const defaultProps = {
+    initialHands: ["👋", "🖐️"],
+    updateHandWaveBadge: mockUpdateHandWaveBadge,
+    hasError: false,
+    onRetry: mockOnRetry,
+    onAdd: mockOnAdd,
+    onDelete: mockOnDelete,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders initial hand buttons", () => {
+    render(<WaveHandPicker {...defaultProps} />);
+    expect(screen.getByText("👋")).toBeInTheDocument();
+    expect(screen.getByText("🖐️")).toBeInTheDocument();
+  });
+
+  it("toggles selection and updates badge", () => {
+    render(<WaveHandPicker {...defaultProps} />);
+    const handButton = screen.getByText("👋");
+    fireEvent.click(handButton);
+    expect(mockUpdateHandWaveBadge).toHaveBeenCalledWith({
+      visible: true,
+      waveText: "👋",
+    });
+
+    fireEvent.click(handButton);
+    expect(mockUpdateHandWaveBadge).toHaveBeenCalledWith({
+      visible: false,
+    });
+  });
+
+  it("calls onAdd when new hand is added", () => {
+    const { container } = render(<WaveHandPicker {...defaultProps} />);
+    const addButton = screen.getByText("➕ Add");
+    fireEvent.click(addButton);
+    const input = container.querySelector(".wave-hand-input");
```

Comment:
Generally, we try to avoid using CSS class names in the unit tests, since these aren't user visible, and we want to test the application from the users perspective.

Usually the best choice for doing the selections is ARIA roles (when they apply), since then we can test both the functionaliy in the unit test and the accessibility at the same time.  Using the text visible to the users on the screen is also OK.

See https://testing-library.com/docs/queries/about/#priority